src/main/java/org/codehaus/plexus/components/secdispatcher/internal/SecUtil.java

```diff
@@ -102,4 +88,64 @@ public static Map<String, String> getConfig(SettingsSecurity sec, String name) {
         }
         return null;
     }
+
+    public static void write(Path target, SettingsSecurity configuration) throws IOException {
+        requireNonNull(target, "file must not be null");
+        requireNonNull(configuration, "sec must not be null");
+        writeFile(target, configuration, false);
+    }
+
+    public static void writeWithBackup(Path target, SettingsSecurity configuration) throws IOException {
```

Comment:
The pattern is usually to have `writeFile(Path, SettingsSecurity)` and `writeFile(Path, SettingsSecutiry, boolean)` methods public methods, the first one calling the second one.  And I don't see any reference to the first method.  If not called, maybe simply keep the one with the backup.

Reply:
test UT uses the "no backup method", but will remove one extra method and apply pattern you propose